pyproject.toml

```diff
@@ -59,7 +59,8 @@ simulators = [
     "projectq>=0.5.0,<0.9.0",
 ]
 cuda = [
-    "cuquantum-python>=23.6.0",
+    "cuquantum-python>=24.03.0",
+    "custatevec>=1.6.0",
     "cupy>=10.4.0",
```

Comment:
One thing that is slightly annoying is that I cannot install this directly from pip in Linux. So `pip install .[all,cuda]` fails. (I could probably figure it out and get cupy, e.g., install via binaries or building from source...) If I install via `conda install -c conda-forge cuquantum-python>=24.03.0`, conda sits there thinking for a very long time. `conda install -c conda-forge cuquantum-python` worked for me and installs the latest version. It also installs both custatevec and cupy. So the extra custatevec and cupy requirements are not needed, at least for conda.

Long story short... I was able to use conda to install and successfully run the associated tests... It feels like maybe not for this PR but in the future we should maybe document how to install (at least reference the appropriate urls). And ideally add a makefile target to set it all up for development.